keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastings.java

```diff
@@ -136,9 +136,10 @@ public void step() {
         }
 
         @Override
-        public void sample(Map<VertexId, List<?>> samplesByVertex) {
+        public void sample(Map<VertexId, List<?>> samplesByVertex, List<Double> logProbs) {
             step();
             takeSamples(samplesByVertex, verticesToSampleFrom);
+            logProbs.add(logProbabilityBeforeStep);
         }
 
         @Override
```

Comment:
Should we be adding support for streaming the samples with an associated log prob too?  It seems for our use case, the streaming stuff might be a much more efficient way of running KL divergence, rather than building up a large set of samples?

Reply:
Agreed. Talked about this with Caleb and that's something we'll add after. Right now individual samples don't have a fitting representation (except the network state, but that isn't specific to samples), so we can't track probabilities.